matcher/solvers/fairir.py

```diff
@@ -0,0 +1,498 @@
+import numpy as np
+import time
+import uuid
+import logging
+import math
+import json
+from .core import SolverException
+
+from .basic_gurobi import Basic
+from gurobipy import *
+
+class FairIR(Basic):
+    """Fair paper matcher via iterative relaxation.
+
+    """
+
+    def __init__(
+        self,
+        minimums,
+        maximums,
+        demands,
+        encoder,
+        thresh=0.0,
+        allow_zero_score_assignments=False,
+        logger=logging.getLogger(__name__)
+        ):
+        """Initialize.
+
+        Args:
+            loads (maximums) - a list of integers specifying the maximum number of papers
+                  for each reviewer.
+            loads_lb (minimums) - a list of ints specifying the minimum number of papers
+                  for each reviewer.
+            coverages (demands) - a list of integers specifying the number of reviews per
+                 paper.
+            weights (stored in encoder) - the affinity matrix (np.array) of papers to reviewers.
+                   Rows correspond to reviewers and columns correspond to
+                   papers.
+
+            Returns:
+                initialized makespan matcher.
+        """
+
+        conflict_sims = encoder.constraint_matrix.T * (encoder.constraint_matrix <= -1).T
+        allowed_sims = encoder.aggregate_score_matrix.transpose() * (encoder.constraint_matrix > -1).T
+        weights = conflict_sims + allowed_sims ## R x P
+
+        self.logger = logger
+        self.n_rev = np.size(weights, axis=0)
+        self.n_pap = np.size(weights, axis=1)
+        self.solved = False
+        self.loads = maximums
+        self.loads_lb = minimums
+        self.coverages = demands
+        self.allow_zero_score_assignments = allow_zero_score_assignments
+        self.weights = weights
+        self.attr_constraints = encoder.attribute_constraints
```

Comment:
these constraints are not part of the constructor in order to keep the same interface with all the other solvers?